darts/models/forecasting/tbats.py

```diff
@@ -0,0 +1,230 @@
+        seasonal_periods
+            Length of each of the periods (amount of observations in each period).
+            TBATS accepts int and float values here.
+            BATS accepts only int values.
+            When ``None`` or empty array, non-seasonal model shall be fitted.
+            If set to ``"freq"``, a single "naive" seasonality
+            based on the series frequency will be used (e.g. [12] for monthly series).
+            In this latter case, the seasonality will be recomputed every time the model is fit.
+        use_arma_errors
+            When True BATS will try to improve the model by modelling residuals with ARMA.
+            Best model will be selected by AIC.
+            If False, ARMA residuals modeling will not be considered.
+        show_warnings
+            If warnings should be shown or not.
+        n_jobs: int, optional (default=None)
+            How many jobs to run in parallel when fitting BATS model.
+            When not provided BATS shall try to utilize all available cpu cores.
+        multiprocessing_start_method: str, optional (default='spawn')
+            How threads should be started.
+            See https://docs.python.org/3/library/multiprocessing.html#contexts-and-start-methods
+        """
+        super().__init__()
+
+        self.kwargs = {
+            "use_box_cox": use_box_cox,
+            "box_cox_bounds": box_cox_bounds,
+            "use_trend": use_trend,
+            "use_damped_trend": use_damped_trend,
+            "seasonal_periods": seasonal_periods,
+            "use_arma_errors": use_arma_errors,
+            "show_warnings": show_warnings,
+            "n_jobs": n_jobs,
+            "multiprocessing_start_method": multiprocessing_start_method,
+        }
+
+        self.seasonal_periods = seasonal_periods
+        self.infer_seasonal_periods = seasonal_periods == "freq"
+        self.model = None
+        np.random.seed(random_state)
+
+    def __str__(self):
+        return "(T)BATS"
+
+    @abstractmethod
+    def _create_model(self):
+        pass
+        # return tbats_TBATS(**self.kwargs)
+
+    def fit(self, series: TimeSeries):
+        super().fit(series)
+        series = self.training_series
+
+        if self.infer_seasonal_periods:
+            self.kwargs["seasonal_periods"] = _seasonality_from_freq(series)
+
+        model = self._create_model()
+        fitted_model = model.fit(series.values())
+        self.model = fitted_model
+
+        return self
+
+    def predict(self, n, num_samples=1):
+        super().predict(n, num_samples)
+
+        yhat = self.model.forecast(steps=n)
+        samples = _compute_samples(self.model, yhat, num_samples)
+
+        return self._build_forecast_series(samples)
+
+    def _is_probabilistic(self) -> bool:
+        return True
+
+    @property
+    def min_train_series_length(self) -> int:
+        if isinstance(self.seasonal_periods, int) and self.seasonal_periods > 1:
```

Comment:
As far as I understand, this will not account for the inferred seasonal periods when `seasonal_periods="freq"` at model creation. Is this intended?

Reply:
It was not intended, good catch!